internal/adminapi/kong_test.go

```diff
@@ -180,130 +162,22 @@ func TestNewKongClientForWorkspace(t *testing.T) {
-	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivateKey.PublicKey, caPrivateKey)
-	if err != nil {
-		t.Errorf("Fail to generate CA certificate %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	caPEM = new(bytes.Buffer)
-	err = pem.Encode(caPEM, &pem.Block{
-		Type:  "CERTIFICATE",
-		Bytes: caBytes,
-	})
-	if err != nil {
-		t.Errorf("Fail to encode CA certificate %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	caPrivateKeyPEM = new(bytes.Buffer)
-	err = pem.Encode(caPrivateKeyPEM, &pem.Block{
-		Type:  "RSA PRIVATE KEY",
-		Bytes: x509.MarshalPKCS1PrivateKey(caPrivateKey),
-	})
-	if err != nil {
-		t.Errorf("Fail to encode CA key %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	cert := &x509.Certificate{
-		SerialNumber: big.NewInt(1658),
-		Subject: pkix.Name{
-			Organization:  []string{"Kong HQ"},
-			Country:       []string{"US"},
-			Province:      []string{"California"},
-			Locality:      []string{"San Francisco"},
-			StreetAddress: []string{"150 Spear Street, Suite 1600"},
-			PostalCode:    []string{"94105"},
-		},
-		IPAddresses:  []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback},
-		NotBefore:    time.Now(),
-		NotAfter:     time.Now().AddDate(0, 0, 1),
-		SubjectKeyId: []byte{1, 2, 3, 4, 6},
-		ExtKeyUsage:  []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
-		KeyUsage:     x509.KeyUsageDigitalSignature,
-	}
-
-	certPrivateKey, err := rsa.GenerateKey(rand.Reader, rsaKeySize)
-	if err != nil {
-		t.Errorf("Fail to generate ingress key %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	certBytes, err := x509.CreateCertificate(rand.Reader, cert, ca, &certPrivateKey.PublicKey, certPrivateKey)
-	if err != nil {
-		t.Errorf("Fail to generate ingress certificate %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	certPEM = new(bytes.Buffer)
-	err = pem.Encode(certPEM, &pem.Block{
-		Type:  "CERTIFICATE",
-		Bytes: certBytes,
-	})
-	if err != nil {
-		t.Errorf("Fail to encode certificate %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	certPrivateKeyPEM = new(bytes.Buffer)
-	err = pem.Encode(certPrivateKeyPEM, &pem.Block{
-		Type:  "RSA PRIVATE KEY",
-		Bytes: x509.MarshalPKCS1PrivateKey(certPrivateKey),
-	})
-	if err != nil {
-		t.Errorf("Fail to encode key %s", err.Error())
-		return nil, nil, nil, err
-	}
-
-	return caPEM, certPEM, certPrivateKeyPEM, nil
-}
-
-func validate(t *testing.T,
-	httpclient *http.Client,
-	caPEM *bytes.Buffer,
-	certPEM *bytes.Buffer,
-	certPrivateKeyPEM *bytes.Buffer,
+func validate(
+	t *testing.T,
+	httpClient *http.Client,
+	caPEM []byte,
+	certPEM []byte,
+	certPrivateKeyPEM []byte,
 	kongAdminToken string,
 ) (err error) {
-	serverCert, err := tls.X509KeyPair(certPEM.Bytes(), certPrivateKeyPEM.Bytes())
+	serverCert, err := tls.X509KeyPair(certPEM, certPrivateKeyPEM)
 	if err != nil {
-		t.Errorf("Fail to load server certificates %s", err.Error())
+		t.Errorf("fail to load server certificates %s", err.Error())
 		return err
 	}
```

Comment:
We can already use the `require.NoError` here, no? Either we this or we return an error for every error that we encounter and not use `t` at all here.